src/Client/Client.fs

```diff
@@ -0,0 +1,63 @@
+namespace TcpEcho
+
+open System
+open System.Buffers
+open System.Net.Sockets
+open System.IO.Pipelines
+open System.Text
+open System.Threading.Tasks
+
+[<AbstractClass>]
+type Client(endpoint: string, port: int) =
+    let minimumBufferSize = 1024
+
+    let mkString (buffer: ReadOnlySequence<byte>) =
+        seq { for segment in buffer -> segment.ToArray() |> Encoding.UTF8.GetString } |> Seq.fold (+) ""
+
+    let rec writeToPipeAsync (writer: PipeWriter) (socket: Socket) = async {
+        let segment = Array.zeroCreate<byte> minimumBufferSize |> ArraySegment
+        let! read = socket.ReceiveAsync(segment, SocketFlags.None) |> Async.AwaitTask
+        match read with
+        | 0 -> return writer.Complete()
+        | bytesRead ->
+            segment.Array.CopyTo(writer.GetMemory(bytesRead))
+            writer.Advance bytesRead
+            let! flusher = writer.FlushAsync().AsTask() |> Async.AwaitTask
+            if flusher.IsCompleted then return writer.Complete()
+
+        match socket.Available with
+        | 0 -> return writer.Complete()
+        | _ -> return! writeToPipeAsync writer socket
+    }
+
+    let rec readFromPipeAsync (reader: PipeReader) str = async {
+        let! result = reader.ReadAsync().AsTask() |> Async.AwaitTask
+        let buffer : ReadOnlySequence<byte> = result.Buffer
+        reader.AdvanceTo(buffer.End, buffer.End)
+
+        let totalString = str + mkString buffer
+        match result.IsCompleted with
+        | true ->
+            reader.Complete()
+            return totalString
+        | false ->
+            return! readFromPipeAsync reader totalString
+    }
+    abstract member Call: string -> Task<string>
+    default __.Call (json: string) =
+        async {
+            use socket = new Socket(SocketType.Stream, ProtocolType.Tcp)
+            do! socket.ConnectAsync(endpoint, port) |> Async.AwaitTask
+
+            let bytes = UTF8Encoding.UTF8.GetBytes(json + Environment.NewLine)
+
+            socket.Send(bytes) |> ignore
+
+            let pipe = Pipe()
+
+            let writer = writeToPipeAsync pipe.Writer socket |> Async.StartAsTask
+            let reader = readFromPipeAsync pipe.Reader "" |> Async.StartAsTask
+
+            do! [ writer :> Task; reader :> Task] |> Task.WhenAll |> Async.AwaitTask
+            return! Async.AwaitTask reader
+        } |> Async.StartAsTask
```

Comment:
let's expose a method returning Async<T> and another one returning Task<T> (which calls to the former by using Async.StartAsTask). This way we would have F#-friendly API and C#-friendly API.

Reply:
Added CallAsyncAsTask, if you have a better naming suggestion tell me :)